graphqlapi/dynamic_generation_meta_fetch.go

```diff
@@ -0,0 +1,650 @@
+		return nil, err
+	}
+	metaClassBooleanPropertyFields, err := genMetaClassBooleanPropertyFields(class, property)
+	if err != nil {
+		return nil, err
+	}
+	metaClassDatePropertyFields, err := genMetaClassDatePropertyFields(class, property)
+	if err != nil {
+		return nil, err
+	}
+	metaClassCRefPropertyFields, err := genMetaClassCRefPropertyObj(class, property)
+	if err != nil {
+		return nil, err
+	}
+
+	switch dataType {
+
+	case schema.DataTypeString:
+		return &graphql.Field{
+			Description: mergeStrings("Meta information about the property \"", property.Name, "\""),
+			Type:        metaClassStringPropertyFields,
+			Resolve: func(p graphql.ResolveParams) (interface{}, error) {
+				return nil, fmt.Errorf("Not supported")
+			},
+		}, nil
+
+	case schema.DataTypeInt:
+		return &graphql.Field{
+			Description: mergeStrings("Meta information about the property \"", property.Name, "\""),
+			Type:        metaClassIntPropertyFields,
+			Resolve: func(p graphql.ResolveParams) (interface{}, error) {
+				return nil, fmt.Errorf("Not supported")
+			},
+		}, nil
+
+	case schema.DataTypeNumber:
+		return &graphql.Field{
+			Description: mergeStrings("Meta information about the property \"", property.Name, "\""),
+			Type:        metaClassNumberPropertyFields,
+			Resolve: func(p graphql.ResolveParams) (interface{}, error) {
+				return nil, fmt.Errorf("Not supported")
+			},
+		}, nil
+
+	case schema.DataTypeBoolean:
+		return &graphql.Field{
+			Description: mergeStrings("Meta information about the property \"", property.Name, "\""),
+			Type:        metaClassBooleanPropertyFields,
+			Resolve: func(p graphql.ResolveParams) (interface{}, error) {
+				return nil, fmt.Errorf("Not supported")
+			},
+		}, nil
+
+	case schema.DataTypeDate:
+		return &graphql.Field{
+			Description: mergeStrings("Meta information about the property \"", property.Name, "\""),
+			Type:        metaClassDatePropertyFields,
+			Resolve: func(p graphql.ResolveParams) (interface{}, error) {
+				return nil, fmt.Errorf("Not supported")
+			},
+		}, nil
+
+	case schema.DataTypeCRef:
+		return &graphql.Field{
+			Description: property.Description,
+			Type:        metaClassCRefPropertyFields,
+			Resolve: func(p graphql.ResolveParams) (interface{}, error) {
+				return nil, fmt.Errorf("Not supported")
+			},
+		}, nil
+
+	default:
+		return &graphql.Field{
+			Description: property.Description,
+			Type:        graphql.String,
+		}, fmt.Errorf(schema.ErrorNoSuchDatatype)
+	}
+}
+
+func genMetaClassStringPropertyFields(class *models.SemanticSchemaClass, property *models.SemanticSchemaClassProperty) (*graphql.Object, error) {
+
+	topOccurencesFields, err := genMetaClassStringPropertyTopOccurrencesFields(class, property)
+	if err != nil {
+		return nil, err
+	}
+	metaFetchMetaPointingFields := graphql.Fields{
+		"type": &graphql.Field{
+			Name:        mergeStrings("Meta", class.Class, "MetaPointingTo"),
+			Description: "datatype of the property",
+			Type:        graphql.Int,
```

Comment:
Is this supposed to be an int?

Reply:
Yes, PointingTo contains only the amount of classes the current class has a connection to